File: Algorithms/Java/2215-find_the_difference_of_two_arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        
        HashMap<Integer, Integer> myMap = new HashMap<>();
        HashSet<Integer> set = new HashSet<>();
        
        for(int i : nums1){
            myMap.put(i, 1);
            set.add(i);
        }

        for(int i: nums2){
            if(set.contains(i)){
                myMap.remove(i);
            }else{
                myMap.put(i, 2);
            }
        }

        ArrayList<Integer> list1 = new ArrayList<>();
        ArrayList<Integer> list2 = new ArrayList<>();
        ArrayList<List<Integer>> finalList = new ArrayList<>();

        for(int i: myMap.keySet()){
            if(myMap.get(i) == 1){
                list1.add(i);
            }else{
                list2.add(i);
            }
        }

        finalList.add(list1);
        finalList.add(list2);

        return finalList;
    }
}

File: Algorithms/Python/2215-find_the_difference_of_two_arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        s1 = set(nums1)
        s2 = set(nums2)

        r1 = set()
        r2 = set()

        for n in nums1:
            if n not in s2:
                r1.add(n)
        
        for n in nums2:
            if n not in s1:
                r2.add(n)
        

        return [list(r1), list(r2)]


        